tests/jobs/test_backfill_job.py

```diff
@@ -79,11 +80,38 @@ def dag_bag():
     return DagBag(include_examples=True)
 
 
-# Patch the MockExecutor into the dict of known executors in the Loader
-@patch.dict(
-    ExecutorLoader.executors, {MOCK_EXECUTOR: f"{MockExecutor.__module__}.{MockExecutor.__qualname__}"}
-)
 class TestBackfillJob:
+    def _mock_executor(self, executor=None):
```

Comment:
Nice

Reply:
Thanks, figuring out how to make this logic accessible as a fixture and as a generator used within tests when mocking needs to happen multiple times was a very long and frustrating couple hours of development haha 😅